### PR TITLE
Adding in missing PUT and DELETE for productListings

### PR DESCRIPTION
**Pull request by bparise**
This change adds in PUT (via create) and DELETE for productListings endpoint

### CODE REVIEW

**Comment by bparise** [comment]
Dope. Forgot test cases to cover this pull request... will do shortly

File: resources/product-listing.js [hunk 1]
```diff
@@ -20,11 +20,35 @@ function ProductListing(shopify) {
...
+/**
+ * Adds the productId to a sales channel
+ *
+ * @param {Number} productId ProductID to add to the sales channel
+ * @return {Promise} Promise that resolves with the result
+ * @public
+ */
+ProductListing.prototype.create = function addProductId(productId) {
+  const url = this.buildUrl(`${productId}`);
+  return this.shopify.request(url, 'PUT', undefined, {})
```
Comment (lpinca) [inline resources/product-listing.js]: The [example](https://help.shopify.com/api/reference/productlisting#create) does not use an empty body. Can we do something like this?

```js
/**
 * Creates a product listing.
 *
 * @param {Number} productId The ID of the product to publish
 * @param {Object} [params] Body parameters
 * @return {Promise} Promise that resolves with the result
 * @public
 */
ProductListing.prototype.create = function create(productId, params) {
  params || (params = { product_id: productId });
  const url = this.buildUrl(productId);
  return this.shopify.request(url, 'PUT', this.key, params || { product_id: productId });
};
```

File: resources/product-listing.js [hunk 2]
```diff
@@ -20,11 +20,35 @@ function ProductListing(shopify) {
...
+};
+
+/**
+ * Deletes the productId to a sales channel
+ *
+ * @param {Number} productId ProductID to delete from the sales channel
+ * @return {Promise} Promise that resolves with the result
+ * @public
+ */
+ProductListing.prototype.delete = function deleteProductId(productId) {
```
Comment (lpinca) [inline resources/product-listing.js]: I think we can remove this as the default implementation https://github.com/MONEI/Shopify-api-node/blob/53b6ac8c1253a46aa423f415bb0c3f7bdd80df95/mixins/base.js#L45-L48 should work.

**Comment by lpinca** [comment]
We also need to update the documentation.

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**